.github/workflows/build.yml

```diff
@@ -0,0 +1,30 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: Install PlatformIO
+        run: |
+          pip install platformio==6.1.13
+
+      - name: Build
+        run: |
+          pio run
```

Comment:
TIL `pio run` will build for all environments but fail overall if one of them fails, so this should catch errors with either 🙌 

```
Environment    Status    Duration
-------------  --------  ------------
espidf         FAILED    00:00:06.156
arduino        SUCCESS   00:01:53.685

1 failed, 1 succeeded in 00:01:59.841 
(penv) ➜  platformio-esp32-espidf git:(main) ✗ echo $?
1
```